include/cuco/detail/static_set/kernels.cuh

```diff
@@ -89,6 +90,257 @@ CUCO_KERNEL void find(InputIt first, cuco::detail::index_type n, OutputIt output
+    bool active_flag = idx < n;
+    auto const active_flushing_tile =
+      cg::binary_partition<flushing_tile_size>(flushing_tile, active_flag);
+    if (active_flag) {
+      auto const found = ref.find(tile, *(first + idx));
+#if defined(CUCO_HAS_CG_INVOKE_ONE)
+      if (found != ref.end()) {
+        cg::invoke_one(tile, [&]() {
+          auto const offset =
+            flushing_counter[flushing_tile_id].fetch_add(1, cuda::std::memory_order_relaxed);
+          flushing_tile_buffer[flushing_tile_id][offset] = cuco::pair{*(first + idx), *found};
+        });
+      }
+#else
+      if (tile.thread_rank() == 0 and found != ref.end()) {
+        auto const offset =
+          flushing_counter[flushing_tile_id].fetch_add(1, cuda::std::memory_order_relaxed);
+        flushing_tile_buffer[flushing_tile_id][offset] = cuco::pair{*(first + idx), *found};
+      }
+#endif
+    }
+
+    flushing_tile.sync();
+    auto const buffer_size =
+      flushing_counter[flushing_tile_id].load(cuda::std::memory_order_relaxed);
+    if (buffer_size + max_matches > flushing_buffer_size) {
+      flush_buffer(flushing_tile,
+                   buffer_size,
+                   flushing_tile_buffer[flushing_tile_id],
+                   counter,
+                   output_probe,
+                   output_match);
+      flushing_tile.sync();
+#if defined(CUCO_HAS_CG_INVOKE_ONE)
+      cg::invoke_one(flushing_tile, [&]() {
+        flushing_counter[flushing_tile_id].store(0, cuda::std::memory_order_relaxed);
+      });
+#else
+      if (flushing_tile.thread_rank() == 0) {
+        flushing_counter[flushing_tile_id].store(0, cuda::std::memory_order_relaxed);
+      }
+#endif
+      flushing_tile.sync();
+    }
+
+    idx += stride;
+  }  // while
+
+  auto const buffer_size = flushing_counter[flushing_tile_id].load(cuda::std::memory_order_relaxed);
+  if (buffer_size > 0) {
+    flush_buffer(flushing_tile,
+                 buffer_size,
+                 flushing_tile_buffer[flushing_tile_id],
+                 counter,
+                 output_probe,
+                 output_match);
+  }
+}
+
+template <typename Size,
+          typename ProbeKey,
+          typename Key,
+          typename AtomicT,
+          typename OutputIt1,
+          typename OutputIt2>
+__device__ void flush_buffer(cooperative_groups::thread_block const& block,
+                             Size buffer_size,
+                             cuco::pair<ProbeKey, Key>* buffer,
+                             AtomicT* counter,
+                             OutputIt1 output_probe,
+                             OutputIt2 output_match)
+{
+  auto i = block.thread_rank();
+  __shared__ Size offset;
+
+#if defined(CUCO_HAS_CG_INVOKE_ONE)
+  cooperative_groups::invoke_one(
+    block, [&]() { offset = counter->fetch_add(buffer_size, cuda::std::memory_order_relaxed); });
+#else
+  if (i == 0) { offset = counter->fetch_add(buffer_size, cuda::std::memory_order_relaxed); }
+#endif
+  block.sync();
```

Comment:
*question*: In the CG `invoke_one` case, is this better written without the explicit `__shared__` offset as:
```
#if defined(CUCO_HAS_CG_INVOKE_ONE)
  Size offset = cg::invoke_one_broadcast(block, [&] { return counter->fetch_add(buffer_size, cuda::std::memory_order_relaxed) });
#else
  __shared__ Size offset;
  if (i == 0) { offset = counter->fetch_add(buffer_size, cuda::std::memory_order_relaxed); }
  block.sync()
#endif
```
?

Reply:
I see your point. `cg::invoke_one_broadcast` only works for tiles but not thread block thus it doesn't work in this particular case. However, your suggestion is valid for numerous other cases in cuco and I will make a PR for this. :+1: Love it.